### PR TITLE
Mainnet v11 upgrade (v11.7.1 software tag)

### PR DESCRIPTION
**Pull request by xlab**
## 1. Overview

Adding a mainnet handler for v11 upgrade.
- extended with Pstake migrations
- added ictest for globalcap on stkxprt
/cc @kruspy @puneet2019

### CODE REVIEW

File: app/upgrades/v11.7.0/upgrades.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	// run x/liquidstake params upgrade to set the defaults in the state
+	lsParams := liquidstaketypes.DefaultParams()
+	lsParams.ModulePaused = false
+	lsParams.AutocompoundFeeRate = sdk.ZeroDec()
+	lsParams.UnstakeFeeRate = sdk.ZeroDec()
+	lsParams.FeeAccountAddress = liquidstaketypes.DummyFeeAccountAcc.String()
+	lsParams.CwLockedPoolAddress = "persistence1md28ykl78fgxtuvj8gvntjlqdamqzx33dr80cf0tkqzx5cv0j8aqtkdjp3"
+
+	// TODO: this to be authz
+	lsParams.WhitelistAdminAddress = "persistence1my083l0vkug8p4yvajpa3dnpf2pwmzmym35lne"
```
Comment (xlab) [inline app/upgrades/v11.7.0/upgrades.go:47]: @puneet2019 please lmk

File: interchaintest/chain_upgrade_test.go [hunk 2]
```diff
@@ -30,9 +30,9 @@ func TestPersistenceUpgradeBasic(t *testing.T) {
 	var (
 		chainName            = "persistence"
 		upgradeRepo          = PersistenceCoreImage.Repository
-		initialVersion       = "v11.5.2"
+		initialVersion       = "v11.6.0"
```
Comment (puneet2019) [inline interchaintest/chain_upgrade_test.go:33]: mainnet is bumping from `v10.4.1-fix`? 
```suggestion
		initialVersion       = "v10.4.1-fix"
```

File: app/upgrades/v11.7.0/upgrades.go (hunk 1 above)
Comment (puneet2019) [inline app/upgrades/v11.7.0/upgrades.go:47]:
```suggestion
	lsParams.WhitelistAdminAddress = "persistence1ealyadcds02yvsn78he4wntt7tpdqhlhg7y2s6"
```

File: app/upgrades/v11.7.0/constants.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package v11_7_0
+
+import (
+	store "github.com/cosmos/cosmos-sdk/store/types"
```
Comment (puneet2019) [inline app/upgrades/v11.7.0/constants.go:4]:
```suggestion
	store "github.com/cosmos/cosmos-sdk/store/types"
	liquidstaketypes "github.com/persistenceOne/pstake-native/v2/x/liquidstake/types"
	ratesynctypes "github.com/persistenceOne/pstake-native/v2/x/ratesync/types"

```

File: app/upgrades/v11.7.0/constants.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const (
+	// UpgradeName defines the on-chain upgrade name.
+	UpgradeName = "v11.7.0"
+)
+
+var Upgrade = upgrades.Upgrade{
+	UpgradeName:          UpgradeName,
+	CreateUpgradeHandler: CreateUpgradeHandler,
+	StoreUpgrades: store.StoreUpgrades{
+		Added: []string{},
```
Comment (puneet2019) [inline app/upgrades/v11.7.0/constants.go:18]:
```suggestion
		Added: []string{liquidstaketypes.StoreKey, ratesynctypes.StoreKey},
```

File: app/upgrades/v11.7.0/upgrades.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+func runLiquidstakeUpgradeMigration(
+	ctx sdk.Context,
+	keepers *keepers.AppKeepers,
+) error {
+	// run x/liquidstake params upgrade to set the defaults in the state
+	lsParams := liquidstaketypes.DefaultParams()
+	lsParams.ModulePaused = false
+	lsParams.AutocompoundFeeRate = sdk.ZeroDec()
+	lsParams.UnstakeFeeRate = sdk.ZeroDec()
+	lsParams.FeeAccountAddress = liquidstaketypes.DummyFeeAccountAcc.String()
```
Comment (puneet2019) [inline app/upgrades/v11.7.0/upgrades.go:43]:
```suggestion
	lsParams.FeeAccountAddress = "persistence1my083l0vkug8p4yvajpa3dnpf2pwmzmym35lne"
```

Comment (puneet2019) [reply app/upgrades/v11.7.0/upgrades.go:43]: even if both fees by design are zero..

**Comment by puneet2019** [comment]
added #296  for pstake migrations

File: interchaintest/chain_upgrade_test.go (hunk 2 above)
Comment (xlab) [reply interchaintest/chain_upgrade_test.go:33]: It's docker tag, so `v10.4.1` but you are right.

File: app/upgrades/v11.7.0/upgrades.go [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+func runLiquidstakeUpgradeMigration(
+	ctx sdk.Context,
+	keepers *keepers.AppKeepers,
+) error {
+	// run x/liquidstake params upgrade to set the defaults in the state
+	lsParams := liquidstaketypes.DefaultParams()
+	lsParams.ModulePaused = false
+	lsParams.AutocompoundFeeRate = sdk.ZeroDec()
+	lsParams.UnstakeFeeRate = sdk.ZeroDec()
+	lsParams.FeeAccountAddress = "persistence1my083l0vkug8p4yvajpa3dnpf2pwmzmym35lne"
```
Comment (puneet2019) [inline app/upgrades/v11.7.0/upgrades.go:47]:
```suggestion
	lsParams.FeeAccountAddress = "persistence1ealyadcds02yvsn78he4wntt7tpdqhlhg7y2s6"
```

File: interchaintest/chain_upgrade_test.go (hunk 2 above)
Comment (puneet2019) [inline interchaintest/chain_upgrade_test.go:33]: not changing this?

File: app/upgrades/v11.7.0/upgrades.go (hunk 6 above)
Comment (puneet2019) [reply app/upgrades/v11.7.0/upgrades.go:47]: `persistenceCore q lsibc params`

File: interchaintest/chain_upgrade_test.go (hunk 2 above)
Comment (xlab) [reply interchaintest/chain_upgrade_test.go]: this is mentioning the outdated codebase.

was fixed in https://github.com/persistenceOne/persistenceCore/pull/295/commits/0f8992ee741b3f9745aec10ef5ef5a65df4b8190 above

**Review by puneet2019 — COMMENTED** [review]
LGTM, just need to update pstake version soon

**Comment by xlab** [comment]
> LGTM, just need to update pstake version soon

here it comes

**Comment by xlab** [comment]
Summary of the `TestLiquidStakeGlobalCapStkXPRT` test:
1) set global cap for LS to 10%
2) try to liquid stake 20% (5M / 25M) - fail
3) try to liquid stake exactly 10% - ok
4) try to liquid stake more - fail
5) liquid-unstake to get some room
6) try to liquid stake more - ok
7) now, delegate (non-liquid stake)
8) try to **stake-to-lp** this delegation - fail
9) liquid-unstake to get more room
10) try to **stake-to-lp** this delegation again - ok
11) accounting that **stake-to-lp** indeed increases liquid staked counter

**Review by puneet2019 — APPROVED** [review]

**Review by kruspy — APPROVED** [review]

**Pull request merged**